### PR TITLE
Fix #1959: Flatten view hierarchy in admin_auth_activity.xml

### LINKED ISSUE #1959: Optimise code in admin_auth_activity.xml
This issue is tracking the optimization work of `admin_auth_activity.xml`

1. Make sure that the layout uses `ConstraintLayout` and there is minimal nesting of views. If this view already uses `ConstraintLayout` then it's fine.

2. Make sure that all TextViews use the style from `styles.xml` file like `Heading1`, `TextFieldLabel`, `Body`, etc. whichever suits best.

While making the above changes the UI should not get affected and the end result should still be the same as before (or an optimized version of UI).

If there are multiple files with the same name, update them all.

## References
Mobile Portrait: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/grid/
Mobile Landscape: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/grid/
Tablet: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/grid/
Wiki: https://github.com/oppia/oppia-android/wiki/Working-on-UI

### PR DESCRIPTION
**Pull request by bladebunny**
## Explanation

Fixes #1959:  Flattens view hierarchy and uses standard styles on text views.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@bladebunny Please make following changes to the PR to make sure that we follow best practices:

1. You have references incorrect issue number in the PR title. Instead of #1949 it should be #1959 .
2. The title of this PR is very general, instead it should be related to what you are working on directly. E.g.: **Fix #1959 : Flatten view hierarchy in `admin_auth_activity.xml` **
3. Ideally you should name branch in all lower case separated by `-`. E.g.: `flatten-admin-auth-activity-view-hierarchy`
4. In PR description, instead of `1959 ` you should write `#1959 `. End Result: `Fixes #1959 . Flattens view hierarchy and uses standard styles on text views.`
5. Also, please fill the checklist in the PR description, you can do that by placing `x` in between the brackets.

File: app/src/main/res/layout-land/admin_auth_activity.xml [hunk 1]
```diff
@@ -39,97 +39,74 @@
...
-            android:textSize="20sp"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintHorizontal_bias="0.0"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent" />
+    <TextView
+      android:id="@+id/admin_auth_heading_textview"
+      style="@style/Heading2"
+      android:layout_width="0dp"
+      android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/admin_auth_activity.xml]: You can remove this and `layout_width`, considering that `Heading2` already has this attribute and you have used `app:layout_constraintHorizontal_bias ="0.0"`

File: app/src/main/res/layout-land/admin_auth_activity.xml [hunk 2]
```diff
@@ -39,97 +39,74 @@
...
-            android:textColor="@color/oppiaAdminAuthSecondaryText"
-            android:textSize="16sp"
-            app:layout_constraintEnd_toEndOf="@+id/admin_auth_heading_textview"
-            app:layout_constraintStart_toStartOf="@+id/admin_auth_heading_textview"
-            app:layout_constraintTop_toBottomOf="@id/admin_auth_heading_textview" />
+    <TextView
+      android:id="@+id/admin_auth_sub_text"
+      style="@style/Subtitle1"
+      android:layout_width="0dp"
+      android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout-land/admin_auth_activity.xml]: Remove `layout_width` and `layout_height` as per above comment.

Also, introduce horizontal bias.

File: app/src/main/res/layout-land/admin_auth_activity.xml [hunk 3]
```diff
@@ -39,97 +39,74 @@
...
-            app:layout_constraintTop_toBottomOf="@id/admin_auth_sub_text"
-            profile:error="@{viewModel.errorMessage}"
-            profile:labelMargin="@{@dimen/space_8dp}" />
+    <org.oppia.android.app.profile.ProfileInputView
+      android:id="@+id/admin_auth_input_pin"
+      android:layout_width="0dp"
+      android:layout_height="wrap_content"
+      android:layout_marginStart="72dp"
+      android:layout_marginTop="24dp"
+      android:layout_marginEnd="72dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/admin_auth_activity.xml]: You have changed the margins, please keep them same as before.

Have a look at the mocks.

https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/a0c84710-3970-4488-beea-6fe6abbdb38a/specs/

File: app/src/main/res/layout-land/admin_auth_activity.xml [hunk 4]
```diff
@@ -39,97 +39,74 @@
...
-      </ScrollView>
+    <Button
+      android:id="@+id/admin_auth_submit_button"
+      style="@style/StateButtonActive"
+      android:background="@{viewModel.isSubmitButtonActive ? @drawable/state_button_primary_background :@drawable/start_button_transparent_background}"
+      android:clickable="@{viewModel.isSubmitButtonActive}"
+      android:minHeight="48dp"
+      android:text="@string/admin_auth_submit"
+      android:textColor="@{viewModel.isSubmitButtonActive ? @color/white : @color/submitButtonInactiveText }"
+      android:layout_marginTop="16dp"
```
Comment (rt4914) [inline app/src/main/res/layout-land/admin_auth_activity.xml]: This margin value should be `44dp` as per mocks.

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 5]
```diff
@@ -40,91 +40,78 @@
...
-          app:layout_constraintEnd_toEndOf="parent"
-          app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toTopOf="parent" />
+    <TextView
+      android:id="@+id/admin_auth_heading_textview"
+      style="@style/Heading2"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginStart="28dp"
+      android:layout_marginTop="12dp"
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: This margin has been updated. It should be kept same as before, `40dp`. Check this mock: https://xd.adobe.com/view/e8aa4198-3940-47f9-514a-f41cc54457f6-9e9b/screen/785546bd-b160-4d9b-b5d6-53ddc1b4a80a/specs/

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 6]
```diff
@@ -40,91 +40,78 @@
...
-          android:textColor="@color/oppiaPrimaryText"
-          android:textSize="20sp"
-          app:layout_constraintEnd_toEndOf="parent"
-          app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toTopOf="parent" />
+    <TextView
+      android:id="@+id/admin_auth_heading_textview"
+      style="@style/Heading2"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: You can remove `layout_width` and `layout_height` and introduce `horizontal_bias="0.0"`

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 7]
```diff
@@ -40,91 +40,78 @@
...
-          android:textColor="@color/oppiaAdminAuthSecondaryText"
-          android:textSize="16sp"
-          app:layout_constraintEnd_toEndOf="parent"
-          app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@id/admin_auth_heading_textview" />
+    <TextView
+      android:id="@+id/admin_auth_sub_text"
+      style="@style/Subtitle1"
+      android:layout_width="0dp"
+      android:layout_height="wrap_content"
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: You can remove `layout_width` and `layout_height` and introduce `horizontal_bias="0.0"`

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 8]
```diff
@@ -40,91 +40,78 @@
...
-          app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@id/admin_auth_sub_text"
-          profile:error="@{viewModel.errorMessage}"
-          profile:labelMargin="@{@dimen/space_8dp}" />
+    <org.oppia.android.app.profile.ProfileInputView
+      android:id="@+id/admin_auth_input_pin"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginStart="28dp"
+      android:layout_marginTop="24dp"
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: This margin has been updated, keep it same as before `48dp`.

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 9]
```diff
@@ -40,91 +40,78 @@
...
-          app:layout_constraintStart_toStartOf="@+id/admin_auth_sub_text"
-          app:layout_constraintTop_toBottomOf="@id/admin_auth_input_pin"
-          app:layout_constraintVertical_bias="0.0" />
-      </androidx.constraintlayout.widget.ConstraintLayout>
+    <Button
+      android:id="@+id/admin_auth_submit_button"
+      style="@style/StateButtonActive"
+      android:layout_marginTop="16dp"
+      android:layout_marginEnd="28dp"
+      android:layout_marginBottom="16dp"
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: There margins have been updated, please keep them same as before.

File: app/src/main/res/layout-land/admin_auth_activity.xml [hunk 10]
```diff
@@ -39,97 +39,74 @@
...
-        android:id="@+id/admin_auth_landscape_scrollView"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:overScrollMode="never"
-        android:scrollbars="none">
-
-        <androidx.constraintlayout.widget.ConstraintLayout
-          android:layout_width="match_parent"
-          android:layout_height="wrap_content"
-          android:paddingBottom="92dp">
```
Comment (rt4914) [inline app/src/main/res/layout-land/admin_auth_activity.xml:60]: We still need this `padding_bottom`, you can introduce this in the root constraint_layout.

**Review by rt4914 — CHANGES_REQUESTED** [review]
In general, the aim of the issue, is to just use `styles` in this file and flatten the view hierarchy without changing anything in the existing UI. Therefore please update the margins everywhere.
Thanks.

**Comment by rt4914** [comment]
Unassigned @BenHenning as this PR will take some time.

**Comment by bladebunny** [comment]
Updated per code review.  I left my original branch name as no point changing that now but will follow the recommendation going forward.

**Comment by rt4914** [comment]
@bladebunny Once you finish the PR, make sure you follow these two steps:
1. Reply to all the comments (DO NOT RESOLVE THEM). This helps us know whether that suggestion was accepted or are there any doubts in that.

2. At this point I think you won't be able to use `Assignees` section and therefore you can mention the reviewer's name.
In this case mine `@rt4914` so that I will directly get notified that I have to look at this PR.

File: app/src/main/res/layout-land/admin_auth_activity.xml (hunk 1 above)
Comment (rt4914) [reply app/src/main/res/layout-land/admin_auth_activity.xml]: @bladebunny This change is pending.

Comment (rt4914) [reply app/src/main/res/layout-land/admin_auth_activity.xml]: Here by this I meant current line which is `layout_height`

File: app/src/main/res/layout-land/admin_auth_activity.xml (hunk 2 above)
Comment (rt4914) [reply app/src/main/res/layout-land/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout-land/admin_auth_activity.xml (hunk 3 above)
Comment (rt4914) [reply app/src/main/res/layout-land/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout-land/admin_auth_activity.xml (hunk 4 above)
Comment (rt4914) [reply app/src/main/res/layout-land/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 5 above)
Comment (rt4914) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 6 above)
Comment (rt4914) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 7 above)
Comment (rt4914) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 8 above)
Comment (rt4914) [reply app/src/main/res/layout/admin_auth_activity.xml]: @bladebunny This is pending

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 9 above)
Comment (rt4914) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done.

File: app/src/main/res/layout-land/admin_auth_activity.xml (hunk 10 above)
Comment (rt4914) [reply app/src/main/res/layout-land/admin_auth_activity.xml:60]: Done.

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 11]
```diff
@@ -14,7 +14,8 @@
     android:id="@+id/admin_auth_container"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@color/addProfileBackground">
+    android:background="@color/addProfileBackground"
+    android:paddingBottom="92dp">
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: @bladebunny Please remove `paddingBottom`

paddingBottom was part of only `landscape-layout` and not `portrait layout`.

**Comment by rt4914** [comment]
@bladebunny Thanks a lot. Great work.

There are still few comments which were not addressed and solved in last PR have a look above.

Also, from [this comment](https://github.com/oppia/oppia-android/pull/1963#issuecomment-706135392) you have done almost everything except step 4. Can you try that again?

Thanks.

File: app/src/main/res/layout-land/admin_auth_activity.xml (hunk 1 above)
Comment (bladebunny) [reply app/src/main/res/layout-land/admin_auth_activity.xml]: Done

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 8 above)
Comment (bladebunny) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 9 above)
Comment (bladebunny) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 11 above)
Comment (bladebunny) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done

**Comment by bladebunny** [comment]
Sorry.  I'm slowly getting the swing of things.  Have updated the bug number in the PR comment; commented on all guidance inline above; and made the latest requested changes.

File: app/src/main/res/layout/admin_auth_activity.xml [hunk 12]
```diff
@@ -40,91 +40,76 @@
...
-          android:textColor="@color/oppiaAdminAuthSecondaryText"
-          android:textSize="16sp"
-          app:layout_constraintEnd_toEndOf="parent"
-          app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@id/admin_auth_heading_textview" />
+    <TextView
+      android:id="@+id/admin_auth_sub_text"
+      style="@style/Subtitle1"
+      android:layout_marginStart="28dp"
+      android:layout_marginTop="48dp"
```
Comment (rt4914) [inline app/src/main/res/layout/admin_auth_activity.xml]: Please have a look at this again.
`layout_marginTop` is `12dp`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Please have a look at before and after screenshot and you will notice that they do not match.

## Tablet-Portrait
1. The margin between two texts is incorrect.
Screenshot 2020-10-15 at 4 26 37 AM

Even this layout is different.
Screenshot_1602715971
Screenshot_1602716144


Also, the mobile-landscape button is getting cut.

Screenshot_1602716132

File: app/src/main/res/layout/admin_auth_activity.xml (hunk 12 above)
Comment (bladebunny) [reply app/src/main/res/layout/admin_auth_activity.xml]: Looking again.  I thought I had matched what was there before.

Comment (bladebunny) [reply app/src/main/res/layout/admin_auth_activity.xml]: Done

**Review by rt4914 — CHANGES_REQUESTED** [review]
Its really nice, there are two issues from previous and new implementation:


In this image you can see submit button has now more margin as compared to before. I check this in tablet design. (Nexus 9)
Screenshot 2020-10-19 at 3 47 19 PM

In mobile landscape if you scroll the layout the toolbar goes out of screen. the toolbar should actually remain constant and should not scroll.
Screenshot 2020-10-19 at 3 48 45 PM

**Comment by rt4914** [comment]
@bladebunny PTAL

**Comment by anandwana001** [comment]
@bladebunny Any update on this?

**Comment by rt4914** [comment]
Closing this as there is no update.

**Pull request closed without merging**